rewrite-java/src/main/java/org/openrewrite/java/typegeneration/TypeInformation.java

```diff
@@ -0,0 +1,265 @@
+                        if (classType.getOwningClass() != null) {
+                            maybeAddType(classType.getOwningClass());
+                            Set<JavaType.Class> innerClasses = innerClassesMap.computeIfAbsent(
+                                    classType.getOwningClass().getFullyQualifiedName(), k -> new HashSet<>());
+                            innerClasses.add(classType);
+                        }
+                        maybeAddType(classType.getSupertype());
+                        if (classType.getInterfaces() != null) {
+                            classType.getInterfaces().forEach(this::maybeAddType);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    public void maybeAddMethod(@Nullable JavaType.Method methodType) {
+
+        if (methodType == null || methodType.getName() == null ||
+                methodType.getDeclaringType() == null || isKnowType(methodType.getDeclaringType())) {
+            return;
+        }
+        maybeAddType(methodType.getDeclaringType());
+        Set<JavaType.Method> methods = typeMap.get(methodType.getDeclaringType());
+        if (methods.contains(methodType)) {
+            return;
+        }
+        methods.add(methodType);
+        JavaType.Method.Signature signature = methodType.getResolvedSignature();
+        if (signature != null) {
+            maybeAddType(signature.getReturnType());
+            if (signature.getParamTypes() != null) {
+                signature.getParamTypes().forEach(this::maybeAddType);
+            }
+        }
+        if (methodType.getTypeParameters() != null) {
+            methodType.getTypeParameters().forEach(this::maybeAddType);
+        }
+    }
+
+    private boolean isKnowType(JavaType.FullyQualified fullyQualified) {
+        return fullyQualified.getPackageName().startsWith("java.");
+    }
+
+    @Nullable private static JavaType.Class resolveClass(JavaType type) {
+        JavaType.Class javaClass = TypeUtils.asClass(type);
+        if (javaClass == null) {
+            JavaType.FullyQualified fq = TypeUtils.asFullyQualified(type);
+            if (fq != null) {
+                javaClass = JavaType.Class.findClass(fq.getFullyQualifiedName());
+            }
+        }
+        return javaClass;
+    }
+
+    List<byte[]> getTypesAsByteCode() {
+
+        Set<JavaType.FullyQualified> filteredTypes = typeMap.keySet().stream().filter(t -> !sourceClasses.contains(t)).collect(Collectors.toSet());
+
+        List<byte[]> classes = new ArrayList<>();
+        for (JavaType.FullyQualified type : filteredTypes) {
+            if (type instanceof JavaType.Class) {
+                classes.add(classToBytes((JavaType.Class) type, typeMap.get(type)));
+            }
+        }
+
+        return classes;
+    }
+
+    byte[] classToBytes(JavaType.Class type, Set<JavaType.Method> methods) {
+        PrintWriter writer = new PrintWriter(System.out);
```

Comment:
This probably shouldn't be using `System.out`, right? Debugging leftover, perhaps?

Reply:
yeah, this was an oversight, I have changed this such that you can pass in a PrintStream during construction to get ASM debugging. Then I changed the TypeGenerationTest to allow this to be toggled on/off.